### PR TITLE
fix(react/twig) - refactor headings to improve accessiblity

### PR DESCRIPTION
**Pull request by Shashika6**
Fixes https://github.com/international-labour-organization/designsystem/issues/865

### Notes :- 

* Currently h3 tags are used in cards which causes issues in accessibility in order of headings

### Solution :- 

* Refactor `h3` tags to be` p` tags by default.
* Add new prop called titleElement if a heading element needs to be added
* Refactored styling of promo card title to h2

### CODE REVIEW

**Comment by justintemps** [comment]
Hey @bashlk in the React package, you need to fix the options provided for Storybook. It should be like a dropdown with a list of the supported elements. Right now it's set to be a generic object which is not right.

image

**Comment by justintemps** [comment]
@Shashika6 the above comment goes for the Detail Card. The Factlist and Multilink Card don't seem to have the option in Storybook, which is a mistake.

**Comment by Shashika6** [comment]
Hey @justintemps added a dropdown to select titleElement options in both twig and react.


Screenshot 2024-04-02 at 21 49 56

File: .changeset/small-brooms-destroy.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@ilo-org/react": patch
+"@ilo-org/styles": patch
+"@ilo-org/twig": patch
+---
+
+Address accessibility issues with heading order in cards by changing `<h3>` tags to `<p>`, Additionaly add an optional prop that allows for configuration of the title element if needed.
```
Comment (justintemps) [inline .changeset/small-brooms-destroy.md:7]: I would change this to:

Address accessibility issues with the heading order of cards by changing the default element used to render the title from an `<h3>` to `<p>`. Additionaly add an optional prop that allows for configuration of the title element if needed.

File: packages/react/src/types/index.ts [hunk 2]
```diff
@@ -64,6 +64,7 @@ export type CardTypes =
   | "factlist"
   | "data";
 
+export type HeadingTypes = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
```
Comment (justintemps) [inline packages/react/src/types/index.ts:67]: @Shashika6 This should include `"p"` if that's the default for cards.

File: packages/react/src/stories/Card/DetailCard.stories.tsx [hunk 3]
```diff
@@ -17,6 +17,10 @@ const CardMeta: Meta<typeof DetailCard> = {
       options: ["wide", "narrow", "fluid"],
       control: { type: "radio" },
     },
+    titleElement: {
+      options: ["h1", "h2", "h3", "h4", "h5"],
```
Comment (justintemps) [inline packages/react/src/stories/Card/DetailCard.stories.tsx:21]: @Shashika6 for all cards, this should include "p" or else the user won't be able to return to the default.

File: packages/twig/src/patterns/components/card_detail/card_detail.wingsuit.yml [hunk 4]
```diff
@@ -69,3 +69,14 @@ detailcard:
       options:
         "true": "true"
         "false": "false"
+    titleElement:
+      type: select
+      label: Title Element
+      description: HTML element used for the title
+      required: false
+      options:
+        h1: h1
```
Comment (justintemps) [inline packages/twig/src/patterns/components/card_detail/card_detail.wingsuit.yml:78]: @Shashika6 here again, the options need to include the default which is "p"

**Review by justintemps — CHANGES_REQUESTED** [review]
Hey @Shashika6 this looks mostly good, we just need to make sure that "p" is included among the options provided to the user.

**Comment by Shashika6** [comment]
Made the changes you requested @justintemps :)

**Comment by justintemps** [comment]
Hey @Shashika6, I added some more things here:

- Changed `titleElement` to `titleLevel` which I think is clearer
- Allowed the `titleLevel` to be set from the Card Group for all cards in the group

Which I was doing the later, I noticed that the Card Group in React still needs to be synced with the Card Group in Twig. There are some settings that are missing (alignment etc.). Could you open a ticket and take care of that?

**Comment by Shashika6** [comment]
Will do @justintemps 🫡

**Pull request merged**